docs/agent/config.md

```diff
@@ -8,6 +8,18 @@ might mean:
  * The option refers to a feature that's currently under development
  * The option refers to a feature that's scheduled but will come later
 
+## Environment variables
+
+All options supported by the Infrastructure Agent in the configuration file can also be set through environment variables;
+ * Option names should be put in uppercase with the `DD_` prefix. (example: `hostname` -> `DD_HOSTNAME`)
+ * Nested variables should be specified with an underscore. (example: `DD_CLUSTER_AGENT_CMD_PORT` -> `cluster_agent.cmd_port`)
+ * List of values should be separated by spaces. (example: `DD_AC_INCLUDE="image:cp-kafka image:k8szk"`)
+ * Any map structure but the proxy settings should be json-formatted. (example: `DD_DOCKER_ENV_AS_TAGS='{ "ENVVAR_NAME": "tag_name" }'`)
+
+Notes:
+ * Giving a nested variables by an environment variable will override all the others nested variables of the parameter specified in the configuration file.
+ * The above note doesn't apply for the proxy setting. (We can specify `DD_PROXY_HTTP` through an environment variable and `proxy.no_proxy` in the configuration file but this is why it is also impossible to set the whole proxy setting through a json.)
```

Comment:
People can use `DD_PROXY_HTTP`, `DD_PROXY_HTTPS` and `DD_PROXY_NO_PROXY` to set the proxy: https://github.com/DataDog/datadog-agent/blob/master/docs/agent/changes.md#proxies

@l0k0ms, @truthbk: could you give a look to the doc update ? Not sure what is the best way to word it.

Reply:
We should wait for the proxy doc to be released: https://github.com/DataDog/documentation/pull/2865 and link to it on the public doc.

Rephrasing would then be: 
r / ```The above note doesn't apply for the proxy setting. (We can specify `DD_PROXY_HTTP` through an environment variable and `proxy.no_proxy` in the configuration file but this is why it is also impossible to set the whole proxy setting through a json.)``` / 

`The above note doesn't apply for the proxy setting. [Refer to the dedicated Agent proxy documentation] to learn more.`

Reply:
Okay, I'll add the above PR as a blocker